cli/config/parser.go

```diff
@@ -165,10 +187,33 @@ func (c *PxfCluster) setGroupProperty(groupName, propertyName, val string) error
 	return nil
 }
 
+// assertNotEmpty checks if a given value is empty and returns the formatted error.
 func assertNotEmpty(val string, errorMsg string) error {
 	if val == "" {
 		return fmt.Errorf("configuration property '%s' has an empty value", errorMsg)
 	}
 	return nil
 }
 
+// createHosts creates a slice of PxfHosts given the hosts file contents.
+func createHosts(clusterName string) ([]*PxfHost, error) {
```

Comment:
```suggestion
func createHosts(clusterName string) ([]*PxfHost, error) {
```
I wonder why pointers: `[]*PxfHost` and not `[]PxfHost`

Reply:
We use a slice of pointers because we wanted uniformity with the maps we hold within `PxfDeployment` and `PxfCluster` struct definitions. Maps have a constraint where if you want to modify the reference within, you need the map to hold pointers. Slices don't have the same constraint but we still use pointers to make it consistent.